src/main/java/soot/dexpler/DexBody.java

```diff
@@ -420,6 +488,10 @@ public Body jimplify(Body b, SootMethod m) {
       int thisRegister = numRegisters - numParameterRegisters - 1;
 
       Local thisLocal = jimple.newLocal("$u" + thisRegister, unknownType); // generateLocal(UnknownType.v());
+      //thisLocal.setType(jBody.getMethod().getDeclaringClass().getType());
+      if (localDebugs.containsKey(thisRegister)) {
+        thisLocal.setName("this");
```

Comment:
We should be careful to avoid duplicate local names in case of obfuscated code with deliberately broken debug info

Reply:
Thanks! It is fixed now by generating a fresh name every time we create a local.